### PR TITLE
Reserve the memory required by MiniMaxH3TEModel to encode images

### PR DESCRIPTION
**Pull request by Haoming02**
#### Problem

When using **MiniMax H3** to do **I2V**, the system often hangs (`1+ min`) at the `MiniMax H3 Image to Video` node. But when doing simple **T2V**, the node finishes instantly (`~5 sec`) instead. This is because the `load_model` only checks for the size of the model, it did not account for the images. So after the model is loaded, encoding the images exceeds the VRAM budget, hence the slow down. 

*(**tip:** using `--reserve-vram 2.5` does help with this problem)*

#### Solution

Use the existing `memory_estimation_function` framework to allocate the memory needed to encode the images *(the `constant` was manually calculated and tested)*

*(**note:** audio and video are not handled)*

#### Result

With this change, when doing **FI2VA**, the `MiniMax H3 Image to Video` node also finishes instantly (`~10 sec`) without having to rely on `--reserve-vram` 

#### Testing Environment

- **GPU:** RTX 4070 Ti S (`16 GB VRAM`)
- **RAM:** 96 GB
- **OS:** Windows 10
- `--disable-dynamic-vram`

### CODE REVIEW

**Comment by Haoming02** [comment]
I have read and agree to the Contributor License Agreement

File: comfy/text_encoders/minimax.py [hunk 1]
```diff
@@ -121,6 +121,21 @@ def __init__(self, device="cpu", dtype=None, model_options={}):
         super().__init__(device=device, dtype=dtype, name="qwen3vl_32b",
                          clip_model=MiniMaxH3ClipModel, model_options=model_options)
 
+    def memory_estimation_function(self, token_weight_pairs, device=None):
+        constant = 0.72
```
[the remark this answers is not in the document]
Comment (Haoming02) [reply comfy/text_encoders/minimax.py]: After more testing, it seems that:
- Processing text alone with no image uses ~1 GB VRAM *(after the `loaded partially;` line got printed, the `Dedicated GPU memory` usage jumps up by another ~1 GB)* *(tested with both short and long prompts)*
- Adding a 1024x1024 image increases the VRAM usage by ~200 MB

I've pushed a new commit reflecting this

[the remark this answers is not in the document]
Comment (Haoming02) [reply comfy/text_encoders/minimax.py]: **Update:** The current data is inconclusive... Running the same test that worked earlier now hangs again... 

Looking at the **Cuda** usage in Task Manager, it sits at 0% with occasional 100% spikes. Memory (RAM) is stable. Disk usage is at 0%. Should I increase the headroom more?

[the remark this answers is not in the document]
Comment (Haoming02) [reply comfy/text_encoders/minimax.py]: Did a more extensive test:

- **Test Environment:** 16 GB Total VRAM ; `--reserve-vram 4.0` to prevent edge case ; `qwen3vl_32b_minimax_h3_nvfp4_awq.safetensors` model is ~14.6 GB on disk
- **Idle**
    - `Dedicated GPU memory` / `Shared GPU memory` = ~1.5 GB / ~0.2 GB
- **After Load ; Before Process**
    - ~11.1 GB / ~4.9 GB
- **Process with 0 Image**
    - ~12.7 GB / ~5.0 GB
- **Process with 1 Image**
    - ~13.5 GB / ~5.1 GB
- **Process with 2 Image**
    - ~13.9 GB / ~5.0 GB

**TL;DR:** The model alone uses ~2.0 GB on average, with each 1024x1024 image using an additional ~400 MB 

Pushed a new commit with comment

**Comment by j2gg0s** [comment]
We hit this in production and arrived at the same fix independently, so here are the measurements in case they help this land.

### Confirming the diagnosis

MiniMax H3 reference-to-video jobs on 22–24 GB cards were failing with `torch.OutOfMemoryError` for roughly a third of that model's attempts, always while loading `MiniMaxH3TEModel`, and always by a tiny margin — median shortfall 18 MiB, none over 1 GiB. Without the hook, `CLIP.load_model` passes `memory_required=0`, so `minimum_memory_required` falls back to `minimum_inference_memory()` (1.28 GB on our pods). The encoder is 27.14 GB, so it is always partially offloaded and the weight budget is `free − 1.28GB` **no matter how much free VRAM there was** — more free memory just loads more of the encoder. That is why the identical graph with identical parameters succeeds or fails at random, and why free-VRAM-at-load does not separate the two groups.

### One detail worth adding: the failing allocation is a weight, not an activation

Every traceback we collected says `Tried to allocate 500.00 MiB`, and 5120 × 25600 × 4 B is exactly 500.0 MiB — one fp32 cast of the largest MLP layer as it streams in under lowvram. So the requirement has a **fixed floor** that is independent of the input, and that floor alone already exceeds 1.28 GB: even a text-only t2v prompt runs on the edge, not just I2V.

### Measurement

On a 4090D with free VRAM ballasted down to match the production L4 (23.27 GB free), the unpatched path reproduces the production failure exactly: `loaded: 21.69GB, offloaded: 5.45GB` → `Tried to allocate 500.00 MiB`. Peak allocation during `encode_token_weights`, above the resident weights:

| positions | 550 | 942 | 1382 | 1966 | 4654 | 8758 | 12862 |
|---|---|---|---|---|---|---|---|
| peak (GB) | 1.27 | 1.36 | 1.52 | 1.86 | 2.29 | 4.55 | 6.93 |

"positions" is the expanded sequence length — what `SDClipModel.process_tokens` actually builds. One image entry is a single token/weight pair but expands to `H*W/1024` embeddings.

**The formula in this PR would have fixed our failures.** For the graph that was crashing (three reference images, 1.43 MPx total, 1966 positions) it reserves 2.76 GB against a measured need of 1.86 GB; at 12862 positions it reserves 7.56 GB against 6.93 GB.

### Two suggestions, offered as refinements rather than objections

1. **Counting expanded positions rather than raw `numel`.** `process_qwen2vl_images` / `process_video_block` round to `patch_size * merge_size` and clamp to `min_pixels`/`max_pixels`, and the vision tower merges `merge_size**2` patches, so a block costs `(h_bar * w_bar) / 1024` positions. Raw `numel` tracks that closely for ordinary reference images, but diverges where the resize policy bites: a video block is 2 frames yet produces one frame's worth of positions (2× over-count), and an image above `max_pixels` is scaled down while `numel` is not. Incidentally this means video *is* charged today, contrary to the note in the description — just at double rate.
2. **The text term is flat**, so a long prompt with no images reserves the same 2 GB as a short one. What we fitted was `1.3 GB + 0.55 MB × positions`, which charges text as well.

Neither is a correctness problem for the image case this PR targets — the two formulas over-reserve relative to each other in different regimes — and over-reserving is nearly free here, since it only moves weights from resident to streamed. We measured 9.0 s patched vs ~10 s unpatched for the same encode, and an end-to-end replay of the failing job (768×768, 362 frames) completed in 11m50s.

### Related

This is the same failure mode as #12867 (`QwenImageTEModel`, still open): a text encoder that does not define `memory_estimation_function` silently gets the `minimum_inference_memory()` floor, and that is only survivable while the encoder comfortably fits in VRAM. Given #12643 and #12686 also had to correct estimates after the fact, would a conservative default on the base class be preferable to per-encoder opt-in?
